### PR TITLE
Convert field editors to DDAU

### PR DESCRIPTION
**Pull request by balinterdi**
See #389

### CODE REVIEW

File: packages/core-types/tests/integration/components/field-editors/date-editor-test.js [hunk 1]
```diff
@@ -1,34 +1,64 @@
...
+        disabled=true
+      }}`);
+    assert.dom('input').isDisabled();
+  });
+
+  test('onchange is called when the field is left', async function(assert) {
+    let changed;
+    this.set('onchange', () => {
+      changed = true;
     });
```
Comment (Turbo87) [inline packages/core-types/tests/integration/components/field-editors/date-editor-test.js]: an alternative pattern is this:
```js
this.set('onchange', () => assert.step('change'));
// ...
assert.verifySteps(['change']);
```

this has the advantage or verifying that the action was only called exactly once

File: packages/core-types/tests/integration/components/field-editors/integer-editor-test.js [hunk 2]
```diff
@@ -40,17 +32,29 @@ module('Integration | Component | field editors/integer editor', function(hooks)
...
+      changed = true;
+    });
+    this.set('model', EmberObject.create({ rating: 3 }));
+    await render(hbs`
+      {{field-editors/integer-editor
+        content=model
+        field="rating"
+        onchange=onchange
+      }}`);
+    await triggerEvent('input', 'blur');
```
Comment (Turbo87) [inline packages/core-types/tests/integration/components/field-editors/integer-editor-test.js]: there is a dedicated [`blur()`](https://github.com/emberjs/ember-test-helpers/blob/master/API.md#blur) helper, which has a bit more code in it to make it work properly on all the browsers

File: packages/core-types/tests/integration/components/field-editors/date-editor-test.js (hunk 1 above)
Comment (balinterdi) [reply packages/core-types/tests/integration/components/field-editors/date-editor-test.js]: I love that. It was passive knowledge but it's so much better.

File: packages/core-types/tests/integration/components/field-editors/integer-editor-test.js (hunk 2 above)
Comment (balinterdi) [reply packages/core-types/tests/integration/components/field-editors/integer-editor-test.js]: Didn't know about this but it'd be great as I've had problem with triggerEvent in the past. For some reason, `await blur('input')` doesn't work, though, it seems like it doesn't trigger the event.

Comment (balinterdi) [reply packages/core-types/tests/integration/components/field-editors/integer-editor-test.js]: Ignore this, the selector (`input`) is not specific enough so it finds another one. Interesting that it has worked with `triggerEvent` though.

Comment (balinterdi) [reply packages/core-types/tests/integration/components/field-editors/integer-editor-test.js]: There's still a timing issue in there. The test fails by default but if I put in a breakpoint in the `blur` method (I did this to check if the event is fired on the right element) it passes.

Comment (balinterdi) [reply packages/core-types/tests/integration/components/field-editors/integer-editor-test.js]: Yep, that works!

File: packages/core-types/tests/integration/components/field-editors/date-editor-test.js [hunk 3]
```diff
@@ -1,34 +1,74 @@
...
 import hbs from 'htmlbars-inline-precompile';
 
+let input;
+
 module('Integration | Component | field editors/date editor', function(hooks) {
   setupRenderingTest(hooks);
 
+  hooks.beforeEach(() => {
+    input = '.field-editor > input';
+  });
```
Comment (Turbo87) [inline packages/core-types/tests/integration/components/field-editors/date-editor-test.js]: why do you set that in `beforeEach`? would be easier to make it `const INPUT = '.field-editor > input';`, right?

File: packages/tools/addon/components/cs-field-editor.js [hunk 4]
```diff
@@ -13,6 +13,13 @@ export default Component.extend({
   enabled: true,
   errors: null,
 
+  init() {
+    this._super(...arguments);
+    if (!this.onchange) {
+      this.onchange = () => {};
+    }
```
Comment (Turbo87) [inline packages/tools/addon/components/cs-field-editor.js]: what's the advantage of this over just defining `onchange() {}` in the class itself?

Comment (Turbo87) [inline packages/tools/addon/components/cs-field-editor.js]: what's the advantage of this over defining `onchange() {}` on the class itself?

Comment (balinterdi) [reply packages/tools/addon/components/cs-field-editor.js]: Nothing, at one point I had a bug that I tried to trace down and forgot to refactor. Thanks!

**Comment by balinterdi** [comment]
@Turbo87 If the tests pass, could you review once more and then merge (if there are no further issues)?

**Pull request merged**